### PR TITLE
[CAZ-2392] Apply logic to payment matrix to filter out dates before a CAZ started charging

### PR DESCRIPTION
**Pull request by dratka**
## Motivation and Context

Jira cards: 
https://eaflood.atlassian.net/browse/CAZ-2392
https://eaflood.atlassian.net/browse/CAZ-2656

## Description

Updated the matrix methods to show only checkboxes to data after the CAZ started charging date. 

## How Has This Been Tested?
I tried to manipulate the date in the CleanAirZone model.

## Screenshots (if appropriate):
More than 6 days:
image

3 days ago:
image

Today:
image

## Checklist:

- [x] It contains only changes required by issue (does not contain other PR)
- [x] Includes link to an issue (if apply)
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: spec/models/clean_air_zones_spec.rb [hunk 1]
```diff
@@ -6,11 +6,13 @@
   subject(:caz) { described_class.new(data) }
 
   let(:data) do
-    { 'name' => name, 'cleanAirZoneId' => id, 'boundaryUrl' => url, 'exemptionUrl' => url }
+    { 'name' => name, 'cleanAirZoneId' => id, 'boundaryUrl' => url, 'exemptionUrl' => url,
```
Comment (likeuwill) [inline spec/models/clean_air_zones_spec.rb:9]: rubocop guidance, each parameter should be from the new line if we can't paste it one line
```    
{ 
 'name' => name, 
 'cleanAirZoneId' => id, 
 'boundaryUrl' => url, 
 'exemptionUrl' => url,
 'activeChargeStartDate' => active_charge_start_date
 }

File: spec/services/payment_dates_spec.rb [hunk 2]
```diff
@@ -3,27 +3,80 @@
...
 
 describe PaymentDates do
-  subject(:service_call) { described_class.call }
+  subject(:service_call) { described_class.call(charge_start_date: charge_start_date) }
 
-  it 'returns next and past dates' do
-    expect(service_call.keys).to contain_exactly(:next, :past)
-  end
+  context 'when charge_start_date was before 6 days ago' do
+    let(:charge_start_date) { Date.today - 10.days }
```
Comment (likeuwill) [inline spec/services/payment_dates_spec.rb:9]: empty line below missing

File: spec/services/payment_dates_spec.rb [hunk 3]
```diff
@@ -3,27 +3,80 @@
...
+          today: true
+        }
+      )
+    end
   end
 
-  it 'returns 6 objects in :past block' do
-    expect(service_call[:past].size).to eq(6)
+  context 'when charge_start_date was before 3 days ago' do
+    let(:charge_start_date) { Date.today - 3.days }
```
Comment (likeuwill) [inline spec/services/payment_dates_spec.rb:34]: same

File: spec/services/payment_dates_spec.rb [hunk 4]
```diff
@@ -3,27 +3,80 @@
...
-  it 'returns a proper object' do
-    expect(service_call[:next].first).to eq(
-      {
-        name: Date.current.strftime(described_class::DISPLAY_DATE_FORMAT),
-        value: Date.current.strftime(described_class::VALUE_DATE_FORMAT),
-        today: true
-      }
-    )
+  context 'when charge_start_date was before today ago' do
+    let(:charge_start_date) { Date.today }
```
Comment (likeuwill) [inline spec/services/payment_dates_spec.rb:59]: same

File: app/services/payment_dates.rb [hunk 5]
```diff
@@ -6,16 +6,27 @@ class PaymentDates < BaseService
...
+    @charge_start_date = charge_start_date
+  end
+
   def call
-    today = Date.current
     {
-      past: ((today - 6.days)..(today - 1.day)).map { |date| parse(date) },
-      next: (today..(today + 6.days)).map { |date| parse(date) }
+      past: past_dates,
+      next: (Date.today..(Date.today + 6.days)).map { |date| parse(date) }
```
Comment (musztardem) [inline app/services/payment_dates.rb:22]: What about future days? e.g. when `activeChargeStartDay` is tomorrow?

Comment (dratka) [reply app/services/payment_dates.rb:22]: Here we will block payments when `activeChargeStartDay` is in the future: https://eaflood.atlassian.net/browse/CAZ-2707

File: app/javascript/styles/application.scss [hunk 6]
```diff
@@ -36,3 +36,21 @@ html {
...
 }
+
+@media (max-width: 40.0525em) {
+  .scrollable-table {
+    overflow: scroll;
+    margin-bottom: 20px;
+  }
+}
+
+
```
Comment (likeuwill) [inline app/javascript/styles/application.scss:47]: extra line

**Pull request merged**